metrics/cli/lib/deploy/deployer.dart

```diff
@@ -0,0 +1,81 @@
+// Use of this source code is governed by the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+import 'package:cli/cli/firebase/firebase_command.dart';
+import 'package:cli/cli/git/git_command.dart';
+import 'package:cli/common/model/services.dart';
+import 'package:cli/deploy/constants/deploy_constants.dart';
+import 'package:cli/flutter/service/flutter_service.dart';
+import 'package:cli/gcloud/service/gcloud_service.dart';
+import 'package:cli/helper/file_helper.dart';
+
+/// A class providing method for deploying the Metrics Web Application.
+class Deployer {
+  /// A service that provides methods for working with Flutter.
+  FlutterService _flutterService;
+
+  /// A service that provides methods for working with GCloud.
+  GCloudService _gcloudService;
+
+  /// A class that provides methods for working with the file system.
+  final FileHelper _fileHelper;
+
+  /// A class that provides methods for working with the Firebase.
+  final FirebaseCommand _firebaseCommand;
+
+  /// A class that provides methods for working with the Git.
+  final GitCommand _gitCommand;
+
+  /// Creates a new instance of the [Deployer] with the given services.
+  ///
+  /// Throws an [ArgumentError] if the given [services] is `null`.
+  /// Throws an [ArgumentError] if the given [FirebaseCommand] is `null`.
+  /// Throws an [ArgumentError] if the given [GitCommand] is `null`.
+  /// Throws an [ArgumentError] if the given [FileHelper] is `null`.
+  Deployer(
+    Services services,
+    this._firebaseCommand,
+    this._gitCommand,
+    this._fileHelper,
+  ) {
```

Comment:
Would it be better to have named parameters? Would it be more readable in initialization?